lib/app/server-listening.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2025, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import process from 'node:process';
+import express from 'express';
+import PromClient from 'prom-client';
+import systemdSocket from 'systemd-socket';
+
+import type {AppArguments} from '../app.interfaces.js';
+import {logger} from '../logger.js';
+
+/**
+ * Starts the web server listening for connections
+ * @param webServer - Express web server
+ * @param appArgs - Application arguments
+ */
+export function startListening(webServer: express.Express, appArgs: AppArguments): void {
+    const ss: {fd: number} | null = systemdSocket(); // TODO: I'm not sure this works any more
```

Comment:
Do we even use the systemd socket in our infra? or is it just a remenant of the past? Maybe it's not worth keeping around if we don't know if it works?

Reply:
It's something someone added once and they seem to want it. The TODO is out of date, I actually checked it and it seems to work just fine.

Reply:
Seems alright to keep around then